import-export-cli/integration/devFirst_test.go

```diff
@@ -122,100 +126,131 @@ func TestInitializeAPIFromAPIDefinitionURL(t *testing.T) {
 		forceFlag: false,
 	}
 
-	output, err := initProjectWithOasFlag(t, args)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	assert.Nil(t, err, "Error while generating Project")
-	assert.Contains(t, output, "Project initialized", "Test InitializeProjectWithDefinitionFlag Failed")
+	validateInitializeProjectWithOASFlag(t, args)
 
 	//Remove Created project and logout
 	base.RemoveDir(projectName)
-	t.Cleanup(func() {
-		base.Execute(t, "logout", apim.GetEnvName())
-	})
+}
+
+func validateImportInitializedProject(t *testing.T, args *initTestArgs) {
+	t.Helper()
+	//Initialize a project with API definition
+	validateInitializeProjectWithOASFlag(t, args)
+
+	time.Sleep(1 * time.Second)
+
+	result, error := importApiFromProject(t, args.initFlag, args.srcAPIM.GetEnvName())
+	assert.Nil(t, error, "Error while importing Project")
+	assert.Contains(t, result, "Successfully imported API", "Error while importing Project")
+}
+
+func validateImportFailedWithInitializedProject(t *testing.T, args *initTestArgs) {
+	t.Helper()
+
+	result, _ := importApiFromProject(t, args.initFlag, args.srcAPIM.GetEnvName())
+	assert.Contains(t, result, "Resource Already Exists", "Test failed because API is imported successfully")
+}
+
+func validateImportUpdatePassedWithInitializedProject(t *testing.T, args *initTestArgs) {
+	t.Helper()
+
+	result, error := importApiFromProjectWithUpdate(t, args.initFlag, args.srcAPIM.GetEnvName())
+	assert.Nil(t, error, "Error while generating Project")
+	assert.Contains(t, result, "Successfully imported API", "Test InitializeProjectWithDefinitionFlag Failed")
 }
 
 //Import API from initialized project with swagger 2 definition
 func TestImportProjectCreatedFromSwagger2Definition(t *testing.T) {
 	apim := apimClients[0]
-	projectName := "OpenAPI3Project"
+	projectName := "Swagger2Project"
+	username := superAdminUser
+	password := superAdminPassword
 
-	output, err := InitializeAPIFromDefinition(t, projectName, utils.TestSwagger2DefinitionPath)
-	assert.Contains(t, output, "Project initialized", "Test InitializeProjectWithDefinitionFlag Failed")
-	assert.Nil(t, err, "Error while generating Project")
+	args := &initTestArgs{
+		ctlUser:   credentials{username: username, password: password},
+		srcAPIM:   apim,
+		initFlag:  projectName,
+		oasFlag:   utils.TestSwagger2DefinitionPath,
+		forceFlag: false,
+	}
 
-	result, error := importApiFromProject(t, projectName, apim.GetEnvName())
-	assert.Nil(t, error, "Error while generating Project")
-	assert.Contains(t, result, "Successfully imported API", "Test InitializeProjectWithDefinitionFlag Failed")
+	//Assert that project import to publisher portal is successful
+	validateImportInitializedProject(t, args)
 
 	//Remove Created project and logout
 	base.RemoveDir(projectName)
```

Comment:
Can we move this _base.RemoveDir()_  into the _validateImportInitializedProject()_ function? It is an essential cleanup step when running this scenario but it is a low level detail that is not required to be known at the high level of the test case.